edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeIdentitiesCache.cs

```diff
@@ -430,15 +458,17 @@ async Task HandleNoServiceIdentity(string id)
         async Task HandleNewServiceIdentity(ServiceIdentity serviceIdentity)
         {
             Option<ServiceIdentity> existing = await this.serviceIdentityHierarchy.Get(serviceIdentity.Id);
-            bool hasUpdated = existing.HasValue && !existing.Contains(serviceIdentity);
-
-            await this.serviceIdentityHierarchy.InsertOrUpdate(serviceIdentity);
-            await this.SaveServiceIdentityToStore(serviceIdentity.Id, new StoredServiceIdentity(serviceIdentity));
-            Events.AddInScope(serviceIdentity.Id);
 
+            bool hasUpdated = await this.serviceIdentityHierarchy.AddOrUpdate(serviceIdentity);
```

Comment:
>hasUpdated [](http://example.com/codeflow?start=17&length=10)

"updated" as a word is getting overloaded here, a better name might be "hasChanged".